### PR TITLE
Fix LocalStack container hostname resolution

### LINKED ISSUE #833: localstack: constructor ignores LOCALSTACK_HOST
The fix to allow wait strategy overrides broke hostname resolution. @cristianrgreco 

_Originally posted by @joebowbeer in https://github.com/testcontainers/testcontainers-node/pull/712#discussion_r1751033935_

The call to `resolveHostname` was moved into the constructor but it should be made after the environment has been assigned. There is no way to provide an environment before the constructor is called so `resolveHostname` will never resolve `LOCALSTACK_HOST`

### PR DESCRIPTION
**Pull request by joebowbeer**
Fixes #833 

Moves the `resolveHostname` invocation back to the `beforeContainerCreated` method.

Fixes the networkAliases override in `resolveHostname` - to now call `withEnvironment`.

Also removes trailing `"` from the `resolveHostname` log message and encloses the value assigned to LOCALSTACK_HOST in quotes. Sample messages:

> testcontainers [INFO] LOCALSTACK_HOST environment variable set to "myhost" (explicitly as environment variable)

> testcontainers [INFO] LOCALSTACK_HOST environment variable set to "myalias" (to match last network alias on container with non-default network)

> testcontainers [INFO] LOCALSTACK_HOST environment variable set to "localhost" (to match host-routable address for container)

### CODE REVIEW

**Comment by cristianrgreco** [comment]
```
FAIL packages/modules/localstack/src/localstack-container.test.ts (47.116 s)
  LocalStackContainer
    ✓ should create a S3 bucket (26507 ms)
    ✓ should use custom network (14459 ms)
    ✕ should assign last network alias to LOCALSTACK_HOST by default (1365 ms)

  ● LocalStackContainer › should assign last network alias to LOCALSTACK_HOST by default

    expect(received).toContain(expected) // indexOf

    Expected substring: "myhost"
    Received string:    ""

      77 |
      78 |     const { output } = await container.exec(["printenv", "LOCALSTACK_HOST"]);
    > 79 |     expect(output).toContain("myhost");
         |                    ^
      80 |
      81 |     await container.stop();
      82 |   });

      at Object.<anonymous> (src/localstack-container.test.ts:79:20)
```

File: packages/modules/localstack/src/localstack-container.test.ts [hunk 1]
```diff
@@ -68,5 +68,19 @@ describe("LocalStackContainer", () => {
...
+
+  it("should not override LOCALSTACK_HOST assignment", async () => {
+    const container = await new LocalstackContainer()
+      .withEnvironment({ LOCALSTACK_HOST: "myhost" })
+      .withNetworkAliases("myalias") // the last alias is assigned to LOCALSTACK_HOST by default
+      .start();
+
+    const { output, exitCode } = await container.exec(["printenv", "LOCALSTACK_HOST"]);
+    expect(exitCode).toBe(0);
+    expect(output).toContain("myhost");
```
Comment (cristianrgreco) [inline packages/modules/localstack/src/localstack-container.test.ts:82]: This is now just testing that the environment variables are passed through. Is there something about this bug that we can test?

Comment (joebowbeer) [reply packages/modules/localstack/src/localstack-container.test.ts:82]: This pass-through is apparently not happening, which may be another bug.

Comment (joebowbeer) [reply packages/modules/localstack/src/localstack-container.test.ts:82]: @cristianrgreco I changed the test to test the original bug, which is that LOCALHOST was being "overridden". 

But now I wonder if the overrides are working...

Comment (joebowbeer) [reply packages/modules/localstack/src/localstack-container.test.ts:82]: Should `resolveHostname` be calling `withEnvironment` to make the assignment?

**Review by cristianrgreco — APPROVED** [review]

**Pull request merged**